### ISSUE TITLE
[core] Fix track length calculation to take into account the ascend/descend

### ISSUE THREAD

**Issue by kirylkaveryn** [organicmaps/organicmaps#9883]
The current method https://github.com/organicmaps/organicmaps/blob/655cf0f174ed78e4163bc160e85cb19289a00a72/geometry/mercator.cpp#L50
is only calculates a projection between points and does not consider the ascend/descend. 
The _true_ length should be the `sqrt(projection ^ 2 + ascend ^ 2)`

The general difference will not be very big but sensitive:

| **Track**       | **projection** | **real length** |
|-----------------|----------------------|----------------------|
| Track 1        | 2146 m               | 2152 m              |
| Track 2        | 6991 m               | 7005 m              |
| Track 3        | 10 002 m             | 10 111 m            |
| Track 4        | 6991 m               | 7005 m              |
| Track 5        | 8158 m               | 8182 m              |
| Track 6        | 15 635 m             | 15 760 m            |

**Comment by pastk** [comment]
Taking it into account for tracks seems fine.

Altering the base `DistanceOnEarth()` which is used everywhere - probably not a good idea.

**Comment by kirylkaveryn** [comment]
> Altering the base `DistanceOnEarth()` which is used everywhere - probably not a good idea.

Right! I updated the issue description that it is only about the track distance calculations.
I created a new method only for the calculation distances between two `PointWithAltitude`
https://github.com/organicmaps/organicmaps/pull/9884

**Comment by Rob-from-VI** [comment]
I have found in the past that recorded GPS tracks displayed with altitude tend to have much bigger variation in altitude that what actually ocurred, with sometime significant ups and downs showing for a track recorded nearly flat.  Thsi would resut in very erroneous distance calculations maybe bigger error than if calculated flat.

I think GPS derived altitudes are somewhat inaccurate in general.

Please check how true are the recorded altitudes before introducing this change.

**Comment by muralito** [comment]
I don't remember the source because i read it several years ago, but I'm almost sure that GPS navigators are optimized for precision in x-y axis (position on the "ground" and not z axis (altitude), and the error in the z-axis can be several times larger than the other axis, so if the altitudes are derived from these tracks this can happen. In the position usually you can expect error to be under 10 m, but in the z axis there can be errors of 50 or 100 meters or even larger. The GPS z axis information is limited by design when used in civil mode, and also when the x-y-z coordinates derived from the satellite positions, are translated to position and altitude, this z is approximated to the ellipsoid in  that position, which is another source for error in altitude.

Luckily for us the error does not propagate to velocity, (not really by luck but because the algorithm to calculate it)
The same could be true when using the elevation for distance calculation, because is the error does not grow, as the algorithm is using the difference in elevation from two adjacent positions (so I suppose that if the positions are near enough, the error cancels itself).

Maybe some volunteers can test it, contributing with readings in level 0 and in the top levels of a building with known height...

So, is not only what you have found or perceived, it is what happens with GPS receivers in navigators, but that probably does not  impact this PR.

@kirylkaveryn : is this method used when the distance is large? did you validate the method when the distance between positions is large? the curvature of the Earth can invalidate your algorithm...

**Comment by Rob-from-VI** [comment]
Our recorded tracks appear to use points about 13m apart horizontally, so it would not take much up and down error to greatly increase the calculated travel distance.

May I suggest an option to show plan-distance alternatively for comparison if this is implemented??

**Comment by kirylkaveryn** [comment]
> the curvature of the Earth can invalidate your algorithm...

The curation is already taken into account in this method.
